raft/src/test/java/org/apache/kafka/raft/internals/RecordsIteratorTest.java

```diff
@@ -110,6 +114,15 @@ public void testCrcValidation(
         // Verify check does not trigger when doCrcValidation is false
         assertDoesNotThrow(() -> testIterator(batches, memRecords, false));
         assertDoesNotThrow(() -> testIterator(batches, fileRecords, false));
+
+        // Fix the corruption
+        memRecords.buffer().putInt(DefaultRecordBatch.CRC_OFFSET, actualCrc);
+
+        // Verify check does not trigger when the corruption is fixed
+        assertDoesNotThrow(() -> testIterator(batches, memRecords, true));
+        FileRecords moreFileRecords = FileRecords.open(TestUtils.tempFile());
```

Comment:
This also applies to existing tests. The tests should close the `FileRecords` objects when they are done using them.